pkg/nodeagent/controller/operatingsystemconfig/containerd_config_test.go

```diff
@@ -0,0 +1,344 @@
+	"github.com/pelletier/go-toml"
+	"github.com/spf13/afero"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/utils/ptr"
+
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/gardener/gardener/pkg/nodeagent/controller/operatingsystemconfig"
+	"github.com/gardener/gardener/pkg/utils/structuredmap"
+)
+
+var (
+	r   operatingsystemconfig.Reconciler
+	osc *extensionsv1alpha1.OperatingSystemConfig
+
+	ctx context.Context
+	log logr.Logger
+)
+
+const (
+	containerdConfigFilePath = "/etc/containerd/config.toml"
+
+	certdsDirValue    = "/etc/containerd/certs.d"
+	importsValue      = "/etc/containerd/conf.d/*.toml"
+	sandBoxImageValue = "foo:1.23"
+	cniBinDirValue    = "/opt/cni/bin"
+)
+
+func init() {
+	ctx = context.Background()
+	log = logr.Discard()
+
+	r = operatingsystemconfig.Reconciler{}
+
+	osc = &extensionsv1alpha1.OperatingSystemConfig{
+		Spec: extensionsv1alpha1.OperatingSystemConfigSpec{
+			CRIConfig: &extensionsv1alpha1.CRIConfig{
+				Name:         extensionsv1alpha1.CRINameContainerD,
+				CgroupDriver: ptr.To(extensionsv1alpha1.CgroupDriverSystemd),
+				Containerd: &extensionsv1alpha1.ContainerdConfig{
+					SandboxImage: sandBoxImageValue,
+				},
+			},
+			Units: []extensionsv1alpha1.Unit{},
+		},
+	}
+}
+
+func getMapEntry(m map[string]any, path structuredmap.Path) (any, error) {
+	if m == nil {
+		return nil, nil
+	}
+
+	key := path[0]
+
+	if len(path) == 1 {
+		return m[key], nil
+	}
+
+	entry, ok := m[key]
+	if !ok {
+		return nil, fmt.Errorf("unable to traverse into data structure because key %q does not exist", key)
+	}
+
+	childMap, ok := entry.(map[string]any)
+	if !ok {
+		return nil, fmt.Errorf("unable to traverse into data structure because value at %q is not a map", key)
+	}
+
+	r, err := getMapEntry(childMap, path[1:])
+	if err != nil {
+		return nil, err
+	}
+
+	return r, nil
+}
+
+func getContainerdConfigValue(fs afero.Afero, path structuredmap.Path) (any, error) {
+	containerdConfigfile, err := fs.ReadFile(containerdConfigFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	containerdConfigContent := map[string]any{}
+
+	if err = toml.Unmarshal(containerdConfigfile, &containerdConfigContent); err != nil {
+		return nil, err
+	}
+
+	return getMapEntry(containerdConfigContent, path)
+}
+
+func loadContainerdConfig(source string, fs afero.Afero) error {
+	containerdConfig, err := os.ReadFile(source)
+	if err != nil {
+		return err
+	}
+
+	err = fs.WriteFile("/etc/containerd/config.toml", containerdConfig, 0644)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ = Describe("containerd configuration file tests", func() {
+
+	BeforeEach(func() {
+		r.FS = afero.Afero{Fs: afero.NewMemMapFs()}
+		Expect(r.FS.MkdirAll("/etc/containerd", 0755)).To(Succeed())
+	})
+
+	Describe("static containerd configuration paths set by gardener-node-agent", func() {
+
+		When("containerd configuration file version is v2", func() {
```

Comment:
Why do we not test v1, but only v2 and v3? If v1 is not of interest anymore, why do we have code dealing with it?

Reply:
The keys in v1 and v2 are the same and did not change, so there is no need to run the tests for v1 and v2 separately - however, I altered the test description to reflect that.